include/z64save.h

```diff
@@ -0,0 +1,199 @@
+#ifndef _Z64SAVE_H_
+#define _Z64SAVE_H_
+
+#include "ultra64.h"
+#include "z64math.h"
+#include <os.h>
+
+typedef struct {
+    /* 0x00 */ u8 buttonItems[4][4];
+    /* 0x10 */ u8 cButtonSlots[4][4];
+    /* 0x20 */ u16 equipment;
+} ItemEquips; // size = 0x0A
+
+typedef struct {
+    /* 0x00 */ u8 items[24];
+    /* 0x18 */ u8 masks[24];
+    /* 0x30 */ s8 ammo[24];
+    /* 0x48 */ u32 upgrades; // some bits are wallet upgrades
+    /* 0x4C */ u32 questItems;
+    /* 0x50 */ u8 dungeonItems[10];
+    /* 0x5A */ s8 dungeonKeys[10];
+    /* 0x64 */ s8 strayFairies[10]; // "orange_fairy"
+    /* 0x6E */ u8 unk_6E[8][3];     // "degnuts_memory_name"
+} Inventory;                        // size = 0x88
+
+typedef struct {
+    /* 0x00 */ s16 scene;
+    /* 0x02 */ Vec3s pos;
+    /* 0x08 */ s16 angle;
+} HorseData; // size = 0x0A
+
+typedef struct {
+    /* 0x0 */ Vec3f pos;
+    /* 0xC */ s16 yaw;
+    /* 0xE */ s16 playerParams;
+    /* 0x10 */ u16 entranceIndex;
+    /* 0x12 */ u8 roomIndex;
+    /* 0x13 */ s8 data;
+    /* 0x14 */ u32 tempSwchFlags;
+    /* 0x18 */ u32 unk_18;
+    /* 0x1C */ u32 tempCollectFlags;
+} RespawnData; // size = 0x20
+
+typedef struct {
+    /* 0x0000 */ u32 entranceIndex; // "scene_no"
+    /* 0x0004 */ u8 equippedMask;   // "player_mask"
+    /* 0x0005 */ u8 unk_05;         // "opening_flag"
+    /* 0x0006 */ u8 unk_06;
+    /* 0x0007 */ u8 linkAge;   // "link_age"
+    /* 0x0008 */ s32 cutscene; // "day_time"
+    /* 0x000C */ u16 time;     // "zelda_time"
+    /* 0x000E */ u16 owlSaveLocation;
+    /* 0x0010 */ u32 isNight;        // "asahiru_fg"
+    /* 0x0014 */ u32 unk_14;         // "change_zelda_time"
+    /* 0x0018 */ u32 day;            // "totalday"
```

Comment:
```suggestion
    /* 0x0010 */ s32 isNight;        // "asahiru_fg"
    /* 0x0014 */ u32 unk_14;         // "change_zelda_time"
    /* 0x0018 */ s32 day;            // "totalday"
```

Reply:
I don't think this is supposed to be signed.

Reply:
https://github.com/zeldaret/mm/blob/cf3eb3b2e19bb60ec9d93b04a57127585e8ac6df/include/z64.h#L649

Reply:
Out of curiosity, I checked out your branch, made the change I pointed out and now it builds OK. You should really try.

Reply:
Oh huh strange, maybe that was a mistake when I made it a u32, let me try that